File: Solutions/0997- Find the Town Judge/Solution.java
class Solution {
    public int findJudge(int n, int[][] trust) {

        int[] following = new int[n + 1]; // who I trust
        int[] follower = new int[n + 1]; // who trust me

        for (int[] item : trust) {
            following[item[0]]++;
            follower[item[1]]++;
        }
        
        for (int i = 1; i < follower.length; i++) {
            if (following[i] == 0 && follower[i] == n - 1)
                return i;
        }
        
        return -1;
    }
}

File: Solutions/0997- Find the Town Judge/Solution.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        following = [0] * (n + 1)  # who I trust
        follower = [0] * (n + 1)  # who trust me

        for item in trust:
            following[item[0]] += 1
            follower[item[1]] += 1

        for i in range(1,n+1):
            if following[i] == 0 and follower[i] == n - 1:
                return i
        
        return -1